libsolidity/lsp/LanguageServer.h

```diff
@@ -113,6 +116,11 @@ class LanguageServer
 	enum class State { Started, Initialized, ShutdownRequested, ExitRequested, ExitWithoutShutdown };
 	State m_state = State::Started;
 
+	// Contains the list of features that are only advertised if advertised by the client
+	// during initialization stage.
+	std::array<std::unique_ptr<HandlerBase>, 3> m_onDemandHandlers;
```

Comment:
Now that I know the context, I think that `experimentalHandlers` would actually be a much better name:
```suggestion
	std::array<std::unique_ptr<HandlerBase>, 3> m_experimentalHandlers;
```

Also, the comment should provide this context and say why we're doing this in the first place. It's probably not a typical LSP behavior.